### PR TITLE
apply aptc to only individual hbx kinds

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186641196

# A brief description of the changes

Current behavior: APTC is being applied to renewal HBX enrollments for consumers that have 2024 eligibility but are enrolled in CoverAll. APTC should not be applied at all.

New behavior: New scope added to HbxEnrollment to help restrict the fetch_enrollments method from grabbing HbxEnrollments that do not have their kind set as 'individual'. Specs added

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

**Review by saikumar9 — CHANGES_REQUESTED** [review]

File: spec/domain/operations/individual/apply_aggregate_to_enrollment_spec.rb [hunk 1]
```diff
@@ -39,21 +39,35 @@
 
   let!(:hbx_profile) {FactoryBot.create(:hbx_profile, :open_enrollment_coverage_period)}
   let!(:person) {FactoryBot.create(:person, :with_consumer_role, :with_active_consumer_role)}
+  let!(:person_coverall) {FactoryBot.create(:person, :with_consumer_role, :with_active_consumer_role)}
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply spec/domain/operations/individual/apply_aggregate_to_enrollment_spec.rb]: https://github.com/ideacrew/enroll/pull/3518/commits/e12cc799811c4845c606d616d49749301a11e7c5#diff-69aa2d91d1a2db8047dd57082ce6c41c89e0b79976d37463aa286e4d50720282R42

File: app/domain/operations/individual/apply_aggregate_to_enrollment.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def validate(params)
 
       def fetch_enrollments_to_renew(eligibility_determination)
         effective_year = eligibility_determination.tax_household.effective_starting_on.year
-        enrollments = eligibility_determination.family.active_household.hbx_enrollments.enrolled_and_renewal.individual_market.by_health.by_year(effective_year)
+        enrollments = eligibility_determination.family.active_household.hbx_enrollments.individual_aptc_only.enrolled_and_renewal.individual_market.by_health.by_year(effective_year)
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply app/domain/operations/individual/apply_aggregate_to_enrollment.rb]: https://github.com/ideacrew/enroll/pull/3518/commits/10e09e2c23190386d3c23c894fc88072d5d3e18b#diff-831ce088fb76833780395638b4b860a4c3c2dcf1099e6d350a7009bfba1b846eR31 && https://github.com/ideacrew/enroll/pull/3518/commits/10e09e2c23190386d3c23c894fc88072d5d3e18b#diff-28b506d6570d53f204d6182db5e6bd01b0c95ecd7b930bdc38592b10d6d6a17dR397-R404

File: app/models/hbx_enrollment.rb [hunk 3]
```diff
@@ -386,6 +386,7 @@ module TermReason
   scope :individual_market,   ->{ where(:kind.nin => ["employer_sponsored", "employer_sponsored_cobra"]) }
   scope :verification_needed, ->{ where(:is_any_enrollment_member_outstanding => true, :aasm_state.in => ENROLLED_STATUSES).or({:terminated_on => nil }, {:terminated_on.gt => TimeKeeper.date_of_record}).order(created_at: :desc) }
   scope :outstanding_enrollments, ->{ individual_market.enrolled.current_year.where(:is_any_enrollment_member_outstanding => true) }
+  scope :individual_aptc_only, ->{ where(:kind.in => ["individual"]) }
```
[the remark this answers is not in the document]
Comment (ATBull81) [reply app/models/hbx_enrollment.rb]: https://github.com/ideacrew/enroll/pull/3518/commits/10e09e2c23190386d3c23c894fc88072d5d3e18b#diff-831ce088fb76833780395638b4b860a4c3c2dcf1099e6d350a7009bfba1b846eR31 && https://github.com/ideacrew/enroll/pull/3518/commits/10e09e2c23190386d3c23c894fc88072d5d3e18b#diff-28b506d6570d53f204d6182db5e6bd01b0c95ecd7b930bdc38592b10d6d6a17dR397-R404 && https://github.com/ideacrew/enroll/pull/3518/files#diff-28b506d6570d53f204d6182db5e6bd01b0c95ecd7b930bdc38592b10d6d6a17dR64-R75

File: app/models/hbx_enrollment.rb [hunk 4]
```diff
@@ -61,11 +61,19 @@ def household=(hh)
...
+  GROUP_KINDS         = %w[employer_sponsored employer_sponsored_cobra].freeze
+  OTHER_KINDS         = %w[unassisted_qhp insurance_assisted_qhp streamlined_medicaid emergency_medicaid hcr_chip].freeze
+
   ENROLLED_AND_RENEWAL_STATUSES = ENROLLED_STATUSES + RENEWAL_STATUSES
 
   ENROLLED_RENEWAL_WAIVED_STATUSES = ENROLLED_STATUSES + RENEWAL_STATUSES + WAIVED_STATUSES
   TERM_REASONS = %w[non_payment voluntary_withdrawl retroactive_canceled].freeze
 
+  IVL_KINDS = [INDIVIDUAL_KIND, COVERALL_KIND].freeze
+  INSURANCE_KINDS = [IVL_KINDS, GROUP_KINDS, OTHER_KINDS].freeze
```
Comment (saikumar9) [inline app/models/hbx_enrollment.rb:75]: @ATBull81 Is this expected? This returns an array of arrays which is not expected.

Comment (ATBull81) [reply app/models/hbx_enrollment.rb]: [1fc80ae](https://github.com/ideacrew/enroll/pull/3518/commits/1fc80aef30e61fda106a5f5a34e2b954d891be59)

**Review by saikumar9 — APPROVED** [review]
lgtm!

**Review by mdkaraman — APPROVED** [review]
Looks good!  Nice job refactoring the "kinds" constants.

**Pull request merged**